opentracing-shim/src/test/java/io/opentelemetry/opentracingshim/SpanBuilderShimTest.java

```diff
@@ -38,6 +40,142 @@ void setUp() {
+
+      try {
+        // The first parent with CHILD_OF becomes the direct parent (if any).
+        SpanData spanData = ((ReadableSpan) childSpan.getSpan()).toSpanData();
+        assertThat(parentSpan2.context().toSpanId()).isEqualTo(spanData.getParentSpanId());
+      } finally {
+        childSpan.finish();
+      }
+    } finally {
+      parentSpan1.finish();
+      parentSpan2.finish();
+      parentSpan3.finish();
+    }
+  }
+
+  @Test
+  void parent_multipleLinks() {
+    SpanShim parentSpan1 = (SpanShim) new SpanBuilderShim(telemetryInfo, SPAN_NAME).start();
+    SpanShim parentSpan2 = (SpanShim) new SpanBuilderShim(telemetryInfo, SPAN_NAME).start();
+    SpanShim parentSpan3 = (SpanShim) new SpanBuilderShim(telemetryInfo, SPAN_NAME).start();
+
+    try {
+      SpanShim childSpan =
+          (SpanShim)
+              new SpanBuilderShim(telemetryInfo, SPAN_NAME)
+                  .addReference(References.FOLLOWS_FROM, parentSpan1.context())
+                  .addReference(References.CHILD_OF, parentSpan2.context())
+                  .addReference(References.FOLLOWS_FROM, parentSpan3.context())
+                  .start();
+
+      try {
+        SpanData spanData = ((ReadableSpan) childSpan.getSpan()).toSpanData();
+        List<LinkData> links = spanData.getLinks();
+        assertThat(links).hasSize(3);
+
+        assertThat(links.get(0).getSpanContext()).isEqualTo(parentSpan1.getSpan().getSpanContext());
+        assertThat(links.get(1).getSpanContext()).isEqualTo(parentSpan2.getSpan().getSpanContext());
+        assertThat(links.get(2).getSpanContext()).isEqualTo(parentSpan3.getSpan().getSpanContext());
+        assertThat(links.get(0).getAttributes().get(SemanticAttributes.OPENTRACING_REF_TYPE))
+            .isEqualTo(SemanticAttributes.OpentracingRefTypeValues.FOLLOWS_FROM);
+        assertThat(links.get(1).getAttributes().get(SemanticAttributes.OPENTRACING_REF_TYPE))
+            .isEqualTo(SemanticAttributes.OpentracingRefTypeValues.CHILD_OF);
+        assertThat(links.get(2).getAttributes().get(SemanticAttributes.OPENTRACING_REF_TYPE))
+            .isEqualTo(SemanticAttributes.OpentracingRefTypeValues.FOLLOWS_FROM);
+
+      } finally {
+        childSpan.finish();
+      }
+    } finally {
+      parentSpan1.finish();
+      parentSpan2.finish();
+      parentSpan3.finish();
+    }
+  }
+
+  @Test
+  void parent_wrongRefType() {
+    SpanShim parentSpan = (SpanShim) new SpanBuilderShim(telemetryInfo, SPAN_NAME).start();
+    try {
+      SpanShim childSpan =
+          (SpanShim)
+              new SpanBuilderShim(telemetryInfo, SPAN_NAME)
+                  .addReference("wrongreftype-value", parentSpan.context())
+                  .start();
+
+      try {
+        // Incorrect typeref values get discarded.
+        SpanData spanData = ((ReadableSpan) childSpan.getSpan()).toSpanData();
+        assertThat(spanData.getParentSpanContext())
+            .isEqualTo(io.opentelemetry.api.trace.SpanContext.getInvalid());
```

Comment:
Should we verify that there are no span links either?

Reply:
Added a check for this.